functions/src/services/GoogleSheet.ts

```diff
@@ -0,0 +1,358 @@
+    spreadsheetId: ISheetId,
+    sheetName: IProjectSpreadSheetNames,
+    columnName: string
+  ): Promise<any>;
+  public abstract findRowWithColumnValue(
+    column: string,
+    rowValueToSearch: string | any,
+    tableToSearch: any
+  ): Promise<any>;
+}
+
+type IRowValues = string[] | number[];
+
+export default class GoogleSheets extends SpreadSheet {
+  protected connection: any;
+
+  protected async connect() {
+    const auth = await gapi.auth.getClient({
+      scopes: [GoogleScopes.SpreadSheets],
+    });
+    this.connection = await gapi.sheets({ version: 'v4', auth });
+    return this.connection;
+  }
+
+  /**
+   * Query for a specific google sheets within a spreadsheet
+   *
+   * @param sheet The sheet to query from google sheets api
+   * @param limit How many rows you want including the header titles
+   */
+  public async getSheet(
+    spreadsheetId: ISheetId,
+    sheetName: IProjectSpreadSheetNames,
+    limit?: number
+  ): Promise<IGetSheetResponse> {
+    await this.connect();
+    // console.log("Range query: ", sheet + this._computeRange(limit));
+    const targetSheet: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: sheetName + this._computeRange(limit),
+    });
+
+    const { statusText, status, data, headers } = targetSheet;
+    if (statusText !== 'OK' || status !== 200) {
+      console.error('Error: Not able to get google sheet');
+      return null;
+    }
+    const { majorDimension, values }: IValueRange = data;
+    if (majorDimension !== IMajorDimensions.Rows || !values || !values.length) {
+      console.error('Error: Values are wrong format');
+      return null;
+    }
+
+    const headerKeys: any[] = values.shift();
+    return {
+      headers: headerKeys,
+      rows: this._convertRows(headerKeys, values),
+    };
+  }
+
+  public async getColumn(
+    spreadsheetId: ISheetId,
+    sheetName: IProjectSpreadSheetNames,
+    columnName: string
+  ) {
+    await this.connect();
+
+    // Retrieve first row of headers
+    const firstRow: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: sheetName + '!A1:Z1',
+    });
+
+    const { statusText, status, data } = firstRow;
+    if (statusText !== 'OK' || status !== 200) {
+      console.error('Error: Not able to get google sheet');
+      return null;
+    }
+
+    const { values }: IValueRange = data;
+    let targetedColumn;
+    const headers: any[] = values[0];
+    headers.forEach((elem, index) => {
+      if (elem === columnName) {
+        targetedColumn = letters[index];
+      }
+    });
+
+    // console.log("Column letter of File Name: ", targetedColumn);
+
+    const entireColumn: any = await this.connection.spreadsheets.values.get({
+      spreadsheetId,
+      majorDimension: IMajorDimensions.Rows,
+      range: sheetName + `!${targetedColumn}1:${targetedColumn}`,
```

Comment:
```suggestion
       range: `${sheetName}!${targetedColumn}:${targetedColumn}`,
```
This might work even without specifying `1`, as `A:A` seems a valid range for spreadsheets.